.github/workflows/python-app.yml

```
@@ -0,0 +1,39 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python application
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
```

```suggestion
      uses: actions/setup-python@v5
```

`actions/setup-python@v3` will likely generate a warning about a deprecated Node.js version.

`v5` is the latest major version of https://github.com/actions/setup-python at the time of this writing.